lua/orgmode/org/indent.lua

```diff
@@ -1,7 +1,88 @@
 local config = require('orgmode.config')
+local headline_lib = require('orgmode.treesitter.headline')
 local ts_utils = require('nvim-treesitter.ts_utils')
 local query = nil
 
+local function get_indent_pad(linenr)
+  local indent_mode = config.org_indent_mode == 'indent'
+  if indent_mode then
+    local headline = headline_lib.from_cursor({ linenr, 0 })
+    if not headline then
+      return 0
+    end
+    return headline:level() + 1
+  end
+  return 0
+end
+
+local function get_indent_for_match(matches, linenr, mode)
```

Comment:
Let's go with early returns here, to separate the logic completely between the matches:

```lua
if match.type == 'headline' then
  return 0
endif

if match.type == 'listitem' then
  -- logic
  return indent
endif

-- etc.
```